rpc/gnmi/gnmi-specification.md

```diff
@@ -1400,8 +1402,10 @@ the current data tree on the server. While the path within the subscription
 SHOULD be a valid path within the set of schema modules that the target
 supports, subscribing to any syntactically valid path within such modules MUST
 be allowed. In the case that a particular path does not (yet) exist, the target
-MUST NOT close the RPC, and instead should continue to monitor for the existence
-of the path, and transmit telemetry updates should it exist in the future.
+would send just a `sync_response` for `ONCE` and `POLL` subscriptions, but MUST
```

Comment:
I think the word "would" ought to be MUST, though I also think this might be clearer to make two different statements for the two different cases, and make it clearer how this applies when you're subscribing to multiple paths, some of which exist and some of which don't. For example, maybe:

> When a `GET` or `POLL` subscription includes paths that do not (yet) exist in the data tree, the target MUST NOT include those paths in its response (for example, a `GET` for exclusively paths that don't exist should return a `SubscriptionResponse` that does not contain any updates).

> When a `STREAM` subscription includes paths that do not (yet) exist in the data tree, the target MUST continue to monitor for the existence of those paths, and transmit telemetry updates should any of them exist in the future. In particular, a `STREAM` subscription consisting _solely_ of paths that are not in the current data tree MUST NOT close the RPC, but instead should continue monitoring in case those paths are created later on.

Reply:
One question: By default, a STREAM starts by sending a notification of the current state; I assume in the case that none of the subscribed paths are set it should still send this notification, it just won't contain any updates?

Reply:
I'm happy with your modification, although I assume by `GET` you mean `ONCE`?

> One question: By default, a STREAM starts by sending a notification of the current state; I assume in the case that none of the subscribed paths are set it should still send this notification, it just won't contain any updates?

This is also my understanding from https://github.com/openconfig/reference/blob/master/rpc/gnmi/gnmi-specification.md#3523-sending-telemetry-updates

Reply:
Oh sorry, yes, I meant ONCE.

Reply:
@gcsl Can you give us your review?